src/main/kotlin/io/unthrottled/doki/promotions/MotivatorPluginPromotionRunner.kt

```diff
@@ -40,17 +41,27 @@ class MotivatorPluginPromotion(
   }
 
   override fun run() {
-    ThemeManager.instance.currentTheme.ifPresent { dokiTheme ->
-      WindowManager.getInstance().suggestParentWindow(
-        ProjectManager.getInstance().openProjects.first()
-      ).toOptional()
-        .ifPresent {
-          MotivatorPromotionDialog(
-            dokiTheme, it, onPromotion
-          ).show()
-        }
-    }
-
+    MotivatorPluginPromotion.runPromotion(onPromotion)
     IdeEventQueue.getInstance().removeIdleListener(this)
   }
+}
+
+object MotivatorPluginPromotion {
+  fun runPromotion(onPromotion: (PromotionResults) -> Unit) {
+    ApplicationManager.getApplication().executeOnPooledThread {
+      ThemeManager.instance.currentTheme.ifPresent { dokiTheme ->
+        WindowManager.getInstance().suggestParentWindow(
+          ProjectManager.getInstance().openProjects.first()
+        ).toOptional()
+          .ifPresent {
+            val promotionAssets = PromotionAssets(dokiTheme)
```

Comment:
We can download the assets here, because this code is being executed on separate thread. 
This means that the UI will not freeze, because when the thread blocks to wait for the IO to complete, it blocks the separate thread and not the UI processing thread.